src/client/dfs/dfs.c

```diff
@@ -3626,6 +3626,14 @@ lookup_rel_path(dfs_t *dfs, dfs_obj_t *root, const char *path, int flags,
 				D_ERROR("daos_array_open() Failed (%d)\n", rc);
 				D_GOTO(err_obj, rc = daos_der2errno(rc));
 			}
+			if (flags & O_TRUNC) {
+				rc = daos_array_set_size(obj->oh, DAOS_TX_NONE, 0, NULL);
+				if (rc) {
+					D_ERROR("Failed to truncate file "DF_RC"\n", DP_RC(rc));
```

Comment:
ok. I will use DHL_ERROR() in next commit. Thank you!

Reply:
hmm why do we need to pass the dfs mountpoint to an error macro? that does not sound OK to me.

Reply:
im fine if we use DL_ERROR here but not DHL_ERROR.
anyway since you are not printing the file name; using D_ERROR is fine and is still not deprecated, and not necessary to switch to the new ones as of yet.

Reply:
Just replaced D_ERROR with DL_ERROR. Thank you!